script/testing/oltpbench/reporting/report_result.py

```diff
@@ -4,22 +4,44 @@
 from oltpbench.reporting.parse_data import parse_data
 from oltpbench.constants import PERFORMANCE_STORAGE_SERVICE_API
 
-def report(api_url, server_data, results_dir, username, password, query_mode='simple'):
-    metadata, timestamp, type, parameters, metrics = parse_data(results_dir)
+
+def report(api_url,
+           server_data,
+           results_dir,
+           username,
+           password,
+           mem_info_dict,
+           query_mode='simple'):
+    metadata, timestamp, benchmark_type, parameters, metrics = parse_data(
+        results_dir)
+    add_mem_info_to_incremental(metrics, mem_info_dict)
     parameters['query_mode'] = query_mode
-    parameters['max_connection_threads'] = server_data['max_connection_threads']
+    parameters['max_connection_threads'] = server_data[
+        'max_connection_threads']
     metadata['environment']['wal_device'] = server_data['wal_device']
     results = {
         'metadata': metadata,
         'timestamp': timestamp,
-        'type': type,
+        'type': benchmark_type,
         'parameters': parameters,
         'metrics': metrics
     }
-    
-    send_results(api_url,username,password,results)
+    send_results(api_url, username, password, results)
 
 
-def send_results(api_url,username,password,results):
-    result = requests.post(api_url + '/oltpbench/', json=results,auth=(username,password))
+def send_results(api_url, username, password, results):
+    result = requests.post(api_url + '/oltpbench/',
+                           json=results,
+                           auth=(username, password))
     result.raise_for_status()
+
+
+def add_mem_info_to_incremental(metrics, mem_info_dict):
```

Comment:
This method seems like it would be more appropriate in the parse_data file

Reply:
done